packages/pigeon/run_tests.sh

```diff
@@ -86,195 +90,337 @@ test_pigeon_android() {
+    curl -L https://github.com/checkstyle/checkstyle/releases/download/checkstyle-8.41/$java_linter >"ci/$java_linter"
+  fi
+  if [ ! -f "ci/$java_formatter" ]; then
+    curl -L https://github.com/google/google-java-format/releases/download/google-java-format-1.3/$java_formatter >"ci/$java_formatter"
+  fi
+  if [ ! -f "ci/$google_checks" ]; then
+    curl -L https://raw.githubusercontent.com/checkstyle/checkstyle/$google_checks_version/src/main/resources/$google_checks >"ci/$google_checks"
+  fi
+  if [ ! -f "ci/$java_error_prone" ]; then
+    curl https://repo1.maven.org/maven2/com/google/errorprone/error_prone_core/2.5.1/$java_error_prone >"ci/$java_error_prone"
+  fi
+  if [ ! -f "ci/$dataflow_shaded" ]; then
+    curl https://repo1.maven.org/maven2/org/checkerframework/dataflow-shaded/3.7.1/$dataflow_shaded >"ci/$dataflow_shaded"
+  fi
+  if [ ! -f "ci/$jformat_string" ]; then
+    curl https://repo1.maven.org/maven2/com/google/code/findbugs/jFormatString/3.0.0/$jformat_string >"ci/$jformat_string"
+  fi
+  if [ ! -f "ci/$javac_jar" ]; then
+    curl https://repo1.maven.org/maven2/com/google/errorprone/javac/9+181-r4173-1/$javac_jar >"ci/$javac_jar"
+  fi
+}
 
-###############################################################################
-# Execute without arguments test
-###############################################################################
-pub run pigeon 1> /dev/null
+run_dart_unittests() {
+  dart analyze bin
+  dart analyze lib
+  dart test
+}
 
-###############################################################################
-# Run unit tests on generated Dart code.
-###############################################################################
-pushd $PWD
-pub run pigeon \
+test_running_without_arguments() {
+  pub run pigeon 1>/dev/null
+}
+
+run_flutter_unittests() {
+  pushd $PWD
+  pub run pigeon \
     --input pigeons/message.dart \
     --dart_out platform_tests/flutter_null_safe_unit_tests/lib/null_safe_pigeon.dart
-cd platform_tests/flutter_null_safe_unit_tests
-flutter pub get
-flutter test test/null_safe_test.dart 
-popd
+  cd platform_tests/flutter_null_safe_unit_tests
+  flutter pub get
+  flutter test test/null_safe_test.dart
+  popd
+}
 
-###############################################################################
-# Mock handler flutter tests.
-###############################################################################
-pushd $PWD
-pub run pigeon \
-  --input pigeons/message.dart \
-  --dart_out mock_handler_tester/test/message.dart \
-  --dart_test_out mock_handler_tester/test/test.dart
-dartfmt -w mock_handler_tester/test/message.dart
-dartfmt -w mock_handler_tester/test/test.dart
-cd mock_handler_tester
-flutter test
-popd
+run_mock_handler_tests() {
+  pushd $PWD
+  pub run pigeon \
+    --input pigeons/message.dart \
+    --dart_out mock_handler_tester/test/message.dart \
+    --dart_test_out mock_handler_tester/test/test.dart
+  dartfmt -w mock_handler_tester/test/message.dart
```

Comment:
why the dartfmt here?

Reply:
Those files can be checked in.  I'm still on the fence of if thats a good idea.  In my PR's I never check them in, but external contributors always do and it makes it easier to review their changes.